Merge from wnprc18.3 svn r.64261 to r.65647

labkey-bpatel opened the pull request:
#### Rationale
Merge from wnprc18.3 svn r.64261 to r.65647 (skipped r. 64864 & 65166 since these were backports)

#### Related Pull Requests
LabKey/commonAssays#200
https://github.com/LabKey/LabDevKitModules/pull/38
LabKey/onprcEHRModules#20
LabKey/platform#1370
LabKey/tnprc_billing#72

labkey-bpatel committed:
Merge from wnprc18.3 svn r.64261-64312

labkey-bpatel committed:
Merge from wnprc18.3 svn r.64313-64598

labkey-bpatel committed:
Merge from wnprc18.3 svn r.64600 to 64828

labkey-bpatel committed:
Merge from wnprc18.3 svn r.64866 to 65149 (skipping r.64864 since its a backport)

labkey-bpatel committed:
Merge from wnprc18.3 svn r.65197 to 65647 (skipping r.65166 since its a backport)

labkey-bpatel committed:
Remove duplicates happened during merge.

labkey-bpatel committed:
Fix merge inconsistencies.
-Delete folders/files that should have been deleted.
-Cleanup/add comments to ehr upgrade scripts.

labkey-jeckels commented on ehr/resources/web/ehr/model/sources/Necropsy.js:
```diff
@@ -1,35 +0,0 @@
-/*
```
This file is still used in ONPRC and SNPRC, so we can't delete it

labkey-bpatel replied:
Thanks for double checking, now restored.

labkey-jeckels commented on ehr/resources/web/ehr/model/sources/Biopsy.js:
```diff
@@ -1,64 +0,0 @@
-/*
```
This file is still used in ONPRC and SNPRC, so we can't delete it

labkey-bpatel replied:
Restored

labkey-jeckels commented on ehr/resources/web/ehr/model/sources/Surgery.js:
```diff
@@ -1,117 +0,0 @@
-/*
```
This file is still used in ONPRC and SNPRC, so we can't delete it

labkey-bpatel replied:
Restored

labkey-jeckels commented on ehr/src/org/labkey/ehr/query/EHRCustomPermissionsTable.java:
```diff
@@ -49,7 +34,6 @@ protected Container getDomainContainer()
         return EHRDomainKind.getDomainContainer(getContainer());
     }
 
-    @Override
```
Why are these lines deleted? Doesn't seem to be a useful change

labkey-bpatel replied:
Not sure, looks like it took what was in wnprc18.3 branch. Added back in.

labkey-jeckels commented on ehr/src/org/labkey/ehr/query/EHRUserSchema.java:
```diff
@@ -24,6 +24,7 @@
 import org.labkey.api.ehr.security.EHRDataAdminPermission;
 import org.labkey.api.ehr.security.EHRProjectEditPermission;
 import org.labkey.api.ehr.security.EHRProtocolEditPermission;
+import org.labkey.api.ehr.security.EHR_BillingAdminPermission;
```
Why is this added? Doesn't seem to be used

labkey-bpatel replied:
removed

labkey-jeckels commented on ehr/src/org/labkey/ehr/security/EHRRequestorRole.java:
```diff
@@ -26,6 +26,7 @@
 import org.labkey.api.ehr.security.EHRRequestPermission;
 import org.labkey.api.ehr.security.EHRRequestSampleDeliveredInsertPermission;
 import org.labkey.api.ehr.security.EHRRequestSampleDeliveredUpdatePermission;
+import org.labkey.api.ehr.security.EHRScheduledInsertPermission;
```
Unused import

labkey-bpatel replied:
removed

labkey-jeckels commented on EHR_ComplianceDB/resources/views/SOPadmin.html:
```diff
@@ -3,6 +3,7 @@
 Ext4.onReady(function(){
     var webpart = <%=webpartContext%>;
     var container = LABKEY.getModuleProperty('ehr_compliancedb', 'EmployeeContainer');
+    var pdfs= LABKEY.getModuleProperty('ehr_compliancedb', 'PDFContainer')
```
Not required for merge, but this and the corresponding .view.xml and .webpart.xml files should be in WNPRC's module, since they're hard-coded to use /WNPRC and don't seem to be used anywhere else.

labkey-bpatel replied:
done! Removed from ehr_compliancedb module, and added to wnprc-modules: https://github.com/WNPRC-EHR-Services/wnprc-modules/tree/20.6_fb_wnprc183_192Merge

labkey-jeckels commented on EHR_ComplianceDB/resources/queries/ehr_compliancedb/SOPrequirements/All SOPs.qview.xml:
```diff
@@ -1,7 +1,7 @@
     <columns>
         <column name="SOP_ID"/>
-        <column name="SOP_ID/Name">
+         <column name="Name">
```
This file should move to WNPRC_EHR since the SOPRequirements moved

labkey-bpatel replied:
Removed from ehr_compliancedb module, and added to wnprc-modules: https://github.com/WNPRC-EHR-Services/wnprc-modules/tree/20.6_fb_wnprc183_192Merge

labkey-jeckels commented on EHR_ComplianceDB/resources/queries/ehr_compliancedb/SOPrequirements/To Read Standard.qview.xml:
```diff
@@ -1,7 +1,7 @@
     <columns>
         <column name="SOP_ID"/>
-        <column name="SOP_ID/Name">
+        <column name="Name">
```
This file should move to WNPRC_EHR since the SOPRequirements moved

labkey-bpatel replied:
Removed from ehr_compliancedb module, and added to wnprc-modules: https://github.com/WNPRC-EHR-Services/wnprc-modules/tree/20.6_fb_wnprc183_192Merge

labkey-jeckels commented on EHR_ComplianceDB/resources/queries/ehr_compliancedb/SOPrequirements/To Read.qview.xml:
```diff
@@ -1,12 +1,12 @@
     <columns>
         <column name="SOP_ID"/>
-        <column name="SOP_ID/Name">
+        <column name="Name">
```
This file should move to WNPRC_EHR since the SOPRequirements moved

labkey-bpatel replied:
Removed from ehr_compliancedb module, and added to wnprc-modules: https://github.com/WNPRC-EHR-Services/wnprc-modules/tree/20.6_fb_wnprc183_192Merge

labkey-jeckels commented on Viral_Load_Assay/resources/web/Viral_Load_Assay/SampleData/LC_data.txt:
```diff
@@ -1,28 +0,0 @@
-TRUE	1	STD_15000000	15.19	1.47E+07	15000000	
```
We should also delete Viral_Load_Assay\src\org\labkey\viral_load_assay\assay\LightCyclerImportMethod.java as it's no longer referenced, and contains a reference to this now deleted file

labkey-bpatel replied:
deleted

labkey-jeckels commented on ehr/resources/web/ehr/grid/Panel.js:
```diff
@@ -49,19 +49,13 @@ Ext4.define('EHR.grid.Panel', {
-        this.getSelectionModel().on('selectionchange', function(sm, models){
-            if (models.length != 1)
-                return;
+        this.getSelectionModel().on('selectionchange', this.handleSectionChangeEvent, this);
```
Looks like a bad merge - the lines below about byPassId should be inside the handleSelectionChangeEvent function below

labkey-bpatel replied:
good eye! thank you! now fixed

labkey-jeckels commented on ehr/resources/web/ehr/form/field/ProjectEntryField.js:
```diff
@@ -10,351 +10,352 @@
-    disabled: false,
-    matchFieldWidth: false,
-    includeDefaultProjects: true,
-    invesLastNameCol: 'lastName',
```
This looks like a problem, as TNPRC is overriding this value and this change appears to completely snip out the investigator value from the template. We should coordinate with WNPRC on the motivation for this change.

labkey-bpatel replied:
I've restored invesLastNameCol for now, will sync with WNPRC

labkey-bpatel replied:
Email sent to client about invesLastNameCol.

labkey-jeckels commented on ehr/src/org/labkey/ehr/EHRController.java:
```diff
@@ -375,8 +376,32 @@ public ModelAndView getView(EHRQueryForm form, BindException errors)
+                else if (isReactForm(form.getSchemaName(), form.getQueryName()))
                 {
+                    //weight is custom since it was introduced in its own controller
+                    if ("weight".equals(form.getQueryName())){
```
We should find an approach that doesn't require hard-coding of WNPRC in general, or of the oddities of the weights form in particular. Could isReactForm() return the URL to use based on the values in the TABLE_FORM_FRAMEWORK_TYPES table?

labkey-jeckels commented on ehr_billing/src/org/labkey/ehr_billing/notification/BillingNotification.java:
```diff
@@ -342,7 +342,7 @@ public void exec(ResultSet object) throws SQLException
                 }
 
-                String financialAnalyst = rs.getString(FieldKey.fromString("investigatorLastName"));
+                String financialAnalyst = rs.getString(FieldKey.fromString("investigator"));
```
@labkey-bpatel  and @labkey-ankurj can you confirm this is OK for both WNPRC and TNPRC?

labkey-ankurj replied:
I am not sure if I've come across/tested runNotification functionality in TNPRC, but a text search for investigatorLastName in both of tnprc repositories has no results.

labkey-ankurj replied:
Looks like there is RequestAdminNotification in tnprc_ehr and ehrModules/EnterDataPanel.js has the key `org.labkey.onprc_ehr.notification.RequestAdminNotification`

labkey-bpatel replied:
WNPRC has been using this in production, so looks like that change is fine.
Thanks @labkey-ankurj for checking in tnprc.

labkey-bpatel committed:
Restore deleted files during merge.

labkey-bpatel committed:
Code review changes:
- Cleanup, optimize imports, fix bad merge
- Restore 'invesLastNameCol' in ProjectEntryField.js.
- Remove unused class LightCycleImportMethod.java

labkey-bpatel committed:
Add space.

labkey-bpatel committed:
Remove 'SOPRequirements' metadata from ehr_compliancedb module, and add to wnprc-modules.

labkey-bpatel committed:
Remove 'SOPAdmin' view from ehr_compliancedb module, and add to wnprc-modules.

labkey-jeckels reviewed (changes requested):
This looks good, pending Chad's revisions to EHRController to avoid hard-coding the React form URLs

labkey-jeckels committed:
Stop hard-coding form URLs and rely on content of metadata table

labkey-jeckels reviewed (approved):
Looks good, pending TeamCity happiness

labkey-jeckels committed:
Point at new package for permission class

labkey-jeckels committed:
Remove unused method to fix build

labkey-jeckels committed:
Fix up schema upgrades, including:

Don't modify EHR schema from WNPRC_EHR scripts

Update getVersion() to getSchemaVersion()

Fix up schema XML files to match expected schema

Fix bootstrapping for WNPRC_EHR upgrades

labkey-jeckels committed:
Various test fixes, including:

Customize source_material query directly instead of trying to mutate a locked FK lookup table

Update to use new assay designer UI

Migrate <javaCustomizer> syntax

labkey-jeckels committed:
More test fixes

labkey-jeckels committed:
Fix SQLServer upgrade scripts

labkey-jeckels committed:
Cascade user delete though ehr_billing.dataaccess

Move code from LabModuleHelper to only place that uses it, ViralLoadAssayTest

Fix rate import codepath in WNPRC_EHRTest

labkey-jeckels committed:
Fix bad merge related to adding QueryWebParts to JS reports

labkey-jeckels committed:
Remove uses of methods injected into object prototypes by Ext3

labkey-jeckels committed:
Remove uses of methods injected into object prototypes by Ext3

labkey-jeckels committed:
Update navMenu from Ext3 to Ext4 and don't pull it into WNPRC with the rest of the Ext3 code

labkey-jeckels committed:
Fully migrate navMenu to ExtJS4

labkey-jeckels committed:
More test fixes

labkey-jeckels committed:
Backport fix from develop branch

labkey-jeckels committed:
Migrate second format() call too

labkey-jeckels committed:
Purge remaining uses of ExtJS3's injected date.format() function

labkey-jeckels committed:
Migrate the .format() calls in HTML files too

labkey-jeckels committed:
Don't rely on ExtJS3 remove method

labkey-jeckels committed:
Update to match HTML tag used for page titles

labkey-jeckels committed:
Avoid validation failure

labkey-jeckels committed:
Split compliance test into ONPRC and WNPRC variants

labkey-jeckels committed:
Update with latest patch from Chad

labkey-jeckels committed:
Get the third format() call hiding at the end of the line

labkey-jeckels committed:
More attempted fixes

labkey-jeckels committed:
Fix typo

labkey-jeckels committed:
Fix ViralLoadAssayTest - restore WNPRC test code, updated to new DOM in 20.x line, and avoiding WNPRC_EHR query metadata if the module is deployed but not enabled in the container
